### ISSUE TITLE
Configuration too restrictive

### ISSUE THREAD

**Issue by Neirda24** [gregurco/GuzzleBundleOAuth2Plugin#8]
As of today `grant_type` takes only a class string. It would be good if it could take a service as well or a way to inject parameters. For example I am trying to plug the OAuth mecanism with auth0. They need an extra parameter wich is called `audience`. We could just integrate it as part of the config but either we do not validate the parameters or we do as said before.

There are plenty of ways to implement the OAuth flow and it seems it can be very tricky to get it done as of today. In the example I gave earlier I have to create a CompilerPass to get the service created dynamically with the `grant_type` class and inject it with the parameter I need.

**Comment by Neirda24** [comment]
Sorry if I am not clear enough. Please feel free to asks any questions that may help you understand the need.

**Comment by gregurco** [comment]
@Neirda24 yep, I got it. Good idea. I will implement it as soon as I will have free time :slightly_smiling_face:

**Comment by Neirda24** [comment]
Hi @gregurco . I saw you implemented it ! Great work.
While waiting I had to find a way to inject it dynamically by extending the `grant_type` option (from `Sainsburys\Guzzle\Oauth2\GrantType\ClientCredentials`) and overiiding the `getAdditionalOptions` as such :

```
    /**
     * {@inheritdoc}
     */
    protected function getAdditionalOptions()
    {
        return [
            'form_params' => [
                'audience' => $this->audience,
            ],
        ];
    }
```

It seems it is causing breaking change.
No worries though we reverted to the original ClientCredential class and used the new option it works perfectly. Just so you know :)

Thank you again !
